### PR TITLE
[ENT-872] Push catalogs to direct integration customers in CSV format (SFTP, SMTP)

### PR DESCRIPTION
**Pull request by UmanShahzad**
**Description:** This PR adds some new fields we'll need to make direct integration for customers through CSV format possible.

**JIRA:** https://openedx.atlassian.net/browse/ENT-872

**Dependencies:** https://github.com/edx/edx-enterprise-data/pull/27

**Merge deadline:** ASAP

**Testing instructions:**

You'll need to install this into your LMS as normal, run the migration, and then follow testing instructions on https://github.com/edx/edx-enterprise-data/pull/27.

**Merge checklist:**

- [ ] Check that the versions of the requirements in the `platform-master.in` file match edx-platform.
- [ ] New requirements are in the right place (`base.in` if only used in enterprise; in the correct `platform-****.in` files if they're hosted in edx-platform)
- [ ] Regenerate requirements with `make upgrade && make requirements` (and make sure to fix any errors).
  **DO NOT** just add dependencies to `requirements/*.txt` files.
- [ ] Called `make static` for webpack bundling if any static content was updated.
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is finished.
- [ ] Delete working branch (if not needed anymore)
- [ ] edx-platform PR (be sure to include edx-platform requirements upgrades that were present in this PR)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @UmanShahzad! I've created [OSPR-2380](https://openedx.atlassian.net/browse/OSPR-2380) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



If you like, you can add yourself to the [AUTHORS](https://github.com/edx/edx-enterprise/blob/master/AUTHORS)      file for this repo, though that isn't required.     Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

**Comment by UmanShahzad** [comment]
@mattdrayer Thanks, I'm glad this will be useful :)

File: enterprise/models.py [hunk 1]
```diff
@@ -1477,6 +1508,7 @@ class EnterpriseCustomerReportingConfiguration(TimeStampedModel):
 
     class Meta:
         app_label = 'enterprise'
+        unique_together = (('enterprise_customer', 'data_type'),)
```
Comment (UmanShahzad) [inline enterprise/models.py]: @mattdrayer One more question: do you want this to be unique together with delivery method and report type as well? This'd allow reporting configs for a single customer for all combinations of data, report, and delivery types. E.g. Progress/Catalog CSV through SFTP/Email, which are 4 combinations, that are all simultaneously possible and will not conflict become of different filename schemes on the delivery end.

Comment (mattdrayer) [reply enterprise/models.py]: That sounds amazingly awesome to me!

**Comment by UmanShahzad** [comment]
@georgebabey Ready for review!

**Comment by UmanShahzad** [comment]
@georgebabey Thanks! I'll add a changelog entry and update the version and merge it, so we can test the corresponding edx-enterprise-data PR.

**Comment by edx-webhook** [comment]
Although this pull request is already merged, I've created [OSPR-5536](https://openedx.atlassian.net/browse/OSPR-5536) so that we can track it in Jira.

There is nothing you have to do. No action is needed from your side.  Thanks again for your contribution.

**Pull request merged**